### PR TITLE
test: added test suite for create_oa_submissions_from_file management command

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** added test suite for create_oa_submissions_from_file management command

JIRA: [AU-514](https://openedx.atlassian.net/browse/AU-514)

**What changed?**

- Tests! 
- Some changes to the management command itself to make it more testable
- **Note**: I'm not bumping version number for this

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/management/commands/create_oa_submissions_from_file.py [hunk 1]
```diff
@@ -224,14 +231,13 @@ def get_display_names(self, submissions_config):
         for ora_config in submissions_config:
             display_name = ora_config['displayName']
             if display_name in display_names:
-                raise CommandError('Duplicate ORA display name found in configuration file:' + display_name)
+                raise CommandError('Duplicate ORA display name found in configuration file: ' + display_name)
             display_names.append(display_name)
         return display_names
 
-    def load_ora_blocks(self, course_id, display_names):
+    def _load_ora_blocks_from_modulestore(self, course_id):
```
Comment (jansenk) [inline openassessment/management/commands/create_oa_submissions_from_file.py:238]: same here, smallest possible mockable untestable unit

File: openassessment/management/tests/test_create_oa_submissions_from_file.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+        super().setUp()
+        for username in [USERNAME_1, USERNAME_2, STAFF_USER_1, STAFF_USER_2, SUPERUSER_USERNAME]:
+            UserFactory.create(username=username)
+
+        self.mock_block = Mock(**MOCK_BLOCK_KWARGS)
+
+        def mock_anonymous_id_for_user(user, course_id):  # pylint: disable=unused-argument
+            return anonymous_user_id(user.username)
+
+        read_config_patcher = patch.object(
```
Comment (jansenk) [inline openassessment/management/tests/test_create_oa_submissions_from_file.py:285]: Did this here rather than inline to avoid having to do four long daisy chained context managers

File: openassessment/management/tests/test_create_oa_submissions_from_file.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+        for reset_user in [USERNAME_1, USERNAME_2]:
+            self.mock_block.clear_student_state.assert_any_call(
+                anonymous_user_id(reset_user),
+                COURSE_ID,
+                str(self.mock_block.location),
+                anonymous_user_id(SUPERUSER_USERNAME)
+            )
+
+
+class CreateSubmissionsFromFileCallCommandTest(TestCase):
```
Comment (jansenk) [inline openassessment/management/tests/test_create_oa_submissions_from_file.py:273]: Is this mostly just duplicating tests above? Yes. Is it still valuable to have a test that actually calls the command and does a simplified check that the side effects happened? I think so

File: openassessment/management/commands/create_oa_submissions_from_file.py [hunk 4]
```diff
@@ -190,18 +193,22 @@ def get_usernames(self, submissions_config):
 
         return learners, course_staff
 
-    def load_anonymous_user_ids(self, course_id, usernames):
-        """
-        Look up all Users with the given usernames, look up their anonymous user ids for the specified course, and
-        store a mapping from username to anonaymous id in self.username_to_anonymous_user_id
-        """
+    def _import_anonymous_id_for_user(self):
```
[the remark this answers is not in the document]
Comment (nsprenkle) [reply openassessment/management/commands/create_oa_submissions_from_file.py:196]: Interesting... I don't really agree with this refactor.

**Reasoning** - I sort of feel like it's *more* bizarre to mock private internals of a class than mocking the imports. The imports are *supposed* to be these separate, modular, things which can be mocked because they exist outside your class under test, whereas mocking an internal function of the class under test inherently ties your tests to the internal structure of the class which makes them brittle and leaks implementation into your test.

File: openassessment/management/tests/test_create_oa_submissions_from_file.py [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+        """ Test for when we are unable to find an anonymous_user_id for a username """
+
+        def mock_anonymous_id_for_user(user, course_id):  # pylint: disable=unused-argument
+            return user.username + "_anonymous"
+
+        for username in [USERNAME_1, USERNAME_2, STAFF_USER_1, STAFF_USER_2, SUPERUSER_USERNAME]:
+            UserFactory.create(username=username)
+
+        expected_error = 'Unable to load anonymous id for user(s) user_not_found'
+        with patch.object(self.cmd, '_import_anonymous_id_for_user', return_value=mock_anonymous_id_for_user):
```
Comment (nsprenkle) [inline openassessment/management/tests/test_create_oa_submissions_from_file.py:213]: Don't like patching class internals. I'm actually surprised that pylint isn't yelling at you for this, it usually does.

**Review by nsprenkle — COMMENTED** [review]
The tests are good but I disagree with the refactor for "testability", I think setting up the mocks on private class under test functions is an antipattern we should avoid.

File: openassessment/management/commands/create_oa_submissions_from_file.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jansenk) [reply openassessment/management/commands/create_oa_submissions_from_file.py:196]: Generally yes - if this was a top level import, and I could do the "mock it where it's used" mock()-ing as intended and just do `mock(...create_oa_submissions_from_file.anonymous_id_for_user)` then yes, I agree.

However, since these are inline imports that won't be able to resolve when running tests isolated from platform, when I say "mocking imports" I don't mean mocking the imported value (sorry, definitely not clear) I mean literally mocking python's import logic. doing a patch over the python import internals. THAT's the hammer.

File: openassessment/management/tests/test_create_oa_submissions_from_file.py (hunk 5 above)
Comment (jansenk) [reply openassessment/management/tests/test_create_oa_submissions_from_file.py:213]: mm, I don't necessarily agree with that, but I can see where you're coming from

**Comment by jansenk** [comment]
https://openedx.atlassian.net/browse/AU-574 opened to address discussions of mocking

**Review by nsprenkle — APPROVED** [review]
Further test refactors slated for https://openedx.atlassian.net/browse/AU-574
